test/replication-luatest/gh_7486_replication_synchro_queue_max_size_test.lua

```diff
@@ -0,0 +1,138 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group('replication_synchro_queue_max_size')
+--
+-- gh-7486: introduce `replication_synchro_queue_max_size`.
+--
+local wait_timeout = 10
+
+local function wait_pair_sync(server1, server2)
+    -- Without retrying it fails sometimes when vclocks are empty and both
+    -- instances are in 'connect' state instead of 'follow'.
+    t.helpers.retrying({timeout = wait_timeout}, function()
+        server1:wait_for_vclock_of(server2)
+        server2:wait_for_vclock_of(server1)
+        server1:assert_follows_upstream(server2:get_instance_id())
+        server2:assert_follows_upstream(server1:get_instance_id())
+    end)
+end
+
+g.before_each(function(cg)
+    t.tarantool.skip_if_not_debug()
+
+    cg.cluster = cluster:new({})
+
+    local box_cfg = {
+        replication = {
+          server.build_listen_uri('master', cg.cluster.id),
+          server.build_listen_uri('replica', cg.cluster.id),
+        },
+        election_mode = 'candidate',
+        replication_synchro_quorum = 2,
+        replication_synchro_timeout = 100000,
+        replication_timeout = 0.1,
+        election_fencing_mode='off',
+    }
+    cg.master = cg.cluster:build_and_add_server({
+        alias = 'master',
+        box_cfg = box_cfg,
+    })
+    box_cfg.election_mode = 'off'
```

Comment:
Seems a little bit dirty. master has `quorum = 2` and election mode 'candidate', but replica has election mode 'off'. Let's make it 'voter'.